packages/spatial/src/physics/classes/Physics.test.ts

```diff
@@ -361,3 +362,714 @@ describe('Physics', () => {
+        Physics.createRigidBody(testEntity, physicsWorld!)
+      })
+
+      afterEach(() => {
+        removeEntity(testEntity)
+        physicsWorld = undefined
+        return destroyEngine()
+      })
+
+      it('should return the correct values', () => {
+        const noBodyEntity = createEntity()
+        assert.equal(Physics.isSleeping(noBodyEntity), true, 'Returns true when the entity does not have a RigidBody')
+        assert.equal(
+          Physics.isSleeping(testEntity),
+          false,
+          "Returns false when the entity is first created and physics haven't been simulated yet"
+        )
+      })
+    })
+
+    describe('setRigidBodyType', () => {
+      let testEntity = UndefinedEntity
+      let physicsWorld: World | undefined = undefined
+
+      beforeEach(async () => {
+        createEngine()
+        await Physics.load()
+        physicsWorld = Physics.createWorld()
+        getMutableState(PhysicsState).physicsWorld!.set(physicsWorld!)
+        physicsWorld!.timestep = 1 / 60
+
+        // Create the entity
+        testEntity = createEntity()
+        setComponent(testEntity, TransformComponent)
+        setComponent(testEntity, RigidBodyComponent, { type: BodyTypes.Dynamic })
+        Physics.createRigidBody(testEntity, physicsWorld!)
+      })
+
+      afterEach(() => {
+        removeEntity(testEntity)
+        physicsWorld = undefined
+        return destroyEngine()
+      })
+
+      it("should assign the correct RigidBodyType to the entity's body", () => {
+        let body = Physics._Rigidbodies.get(testEntity)!
+        assert.equal(body.bodyType(), RigidBodyType.Dynamic)
+        // Check change to fixed
+        Physics.setRigidBodyType(testEntity, BodyTypes.Fixed)
+        body = Physics._Rigidbodies.get(testEntity)!
+        assert.notEqual(body.bodyType(), RigidBodyType.Dynamic, "The RigidBody's type was not changed")
+        assert.equal(body.bodyType(), RigidBodyType.Fixed, "The RigidBody's type was not changed to Fixed")
+        // Check change to dynamic
+        Physics.setRigidBodyType(testEntity, BodyTypes.Dynamic)
+        body = Physics._Rigidbodies.get(testEntity)!
+        assert.notEqual(body.bodyType(), RigidBodyType.Fixed, "The RigidBody's type was not changed")
+        assert.equal(body.bodyType(), RigidBodyType.Dynamic, "The RigidBody's type was not changed to Dynamic")
+        // Check change to kinematic
+        Physics.setRigidBodyType(testEntity, BodyTypes.Kinematic)
+        body = Physics._Rigidbodies.get(testEntity)!
+        assert.notEqual(body.bodyType(), RigidBodyType.Dynamic, "The RigidBody's type was not changed")
+        assert.equal(
+          body.bodyType(),
+          RigidBodyType.KinematicPositionBased,
+          "The RigidBody's type was not changed to KinematicPositionBased"
+        )
+      })
+    })
+
+    describe('setRigidbodyPose', () => {
+      const position = new Vector3(1, 2, 3)
+      const rotation = new Quaternion(4, 5, 6, 1)
```

Comment:
we should only be using valid values, we should normalize this value

Reply:
The passed quaternion was being normalized internally by Rapier.
So the issue was that I couldn't check the Expected and the Result, because they were different no matter the expected value.
Calling `.normalize()` on the input/expected quaternion, no matter their values, solved the problem :writing_hand: